### ISSUE TITLE
CodeKit won't start

### ISSUE THREAD

**Issue by steveblackwood** [bdkjones/CodeKit-1-Old#629]
I can't start CodeKit on my Mac mini. It used to work fine, so I deleted it, and installed a fresh copy. Still get the same errors, first the one about a perm problem:
![codekit-error-1](https://f.cloud.github.com/assets/1672848/735839/7ebe1fde-e302-11e2-88c9-069724105fb9.png)
And after clicking "OK", I get this error:
![codekit-error-2](https://f.cloud.github.com/assets/1672848/735842/8391d88e-e302-11e2-9aaf-70a491d8fcfc.png)
Any ideas?

**Comment by steveblackwood** [comment]
Additional information:

Found these two entries in Console.app:

7/2/13 5:36:39.832 AM CodeKit[862]: +entityForName: nil is not a legal NSManagedObjectContext parameter searching for entity name 'LPGroup'

7/2/13 5:36:39.834 AM CodeKit[862]: (
    0   CoreFoundation                      0x00007fff9431cb06 __exceptionPreprocess + 198
    1   libobjc.A.dylib                     0x00007fff8e38b3f0 objc_exception_throw + 43
    2   CoreData                            0x00007fff93fcbd92 +[NSEntityDescription entityForName:inManagedObjectContext:] + 98
    3   CodeKit                             0x00000001072d11ca -[LPModelController initializeModelGroups] + 183
    4   CodeKit                             0x00000001072d0dfa -[LPModelController init] + 468
    5   AppKit                              0x00007fff8c5abdea -[NSCustomObject nibInstantiate] + 382
    6   AppKit                              0x00007fff8c58b652 -[NSIBObjectData instantiateObject:] + 311
    7   AppKit                              0x00007fff8c58adf7 -[NSIBObjectData nibInstantiateWithOwner:topLevelObjects:] + 337
    8   AppKit                              0x00007fff8c56a11d loadNib + 317
    9   AppKit                              0x00007fff8c569649 +[NSBundle(NSNibLoading) _loadNibFile:nameTable:withZone:ownerBundle:] + 219
    10  AppKit                              0x00007fff8c6bc58c +[NSBundle(NSNibLoading) loadNibFile:externalNameTable:withZone:] + 140
    11  AppKit                              0x00007fff8c6ef5db -[NSWindowController loadWindow] + 199
    12  AppKit                              0x00007fff8c6ef2b5 -[NSWindowController window] + 77
    13  AppKit                              0x00007fff8c72fb79 -[NSWindowController showWindow:] + 53
    14  CodeKit                             0x00000001072ce1d6 -[LessPlusAppDelegate applicationWillFinishLaunching:] + 438
    15  CoreFoundation                      0x00007fff942ceeda _CFXNotificationPost + 2554
    16  Foundation                          0x00007fff936477b6 -[NSNotificationCenter postNotificationName:object:userInfo:] + 64
    17  AppKit                              0x00007fff8c5c14a4 -[NSApplication finishLaunching] + 331
    18  AppKit                              0x00007fff8c5c101a -[NSApplication run] + 124
    19  AppKit                              0x00007fff8c565bd6 NSApplicationMain + 869
    20  CodeKit                             0x00000001072ccefe main + 114
    21  CodeKit                             0x00000001072cce84 start + 52
    22  ???                                 0x0000000000000002 0x0 + 2
)

**Comment by bdkjones** [comment]
Hi Steve,

```
CodeKit is failing to launch because it can't save its data file to the standard location. It looks like the permissions or ACL (access control list) for your ~/Library folder are messed up. Try repairing your disk permissions with Disk Utility to see if that resolves the issue.
```

-Bryan

**Comment by steveblackwood** [comment]
I should have mentioned that I already had done that, without success. However, I added my user account with r+w access to that dir long enough for CodeKit to create the file, then reset perms to default, and now it's working fine. I guess once it gets access, it keeps it.

Thanks!

**Comment by bdkjones** [comment]
Hi Steve,

If you're modifying the permissions on ~/Library as a regular habit, you're doing something wrong. Your user account should _always_ have read/write permissions for that folder and all subfolders in it. If you've changed that manually, you're going to break lots of stuff, not just CodeKit.

**Comment by steveblackwood** [comment]
I've never changed that folder, _ever_. CodeKit has been the only app to ever trigger that error. I tried fixing perms with the disk utility, and that didn't work (found no errors), so I don't know what else I could have done. The perms now (and before the CodeKit error) show:

system: read & write
wheel: read only
everyone: read only

Is that not what it should be? To fix the error I was getting, I added (me), then removed it once finished.

Steve
